internal/apiresource/buildconfig.go

```diff
@@ -0,0 +1,184 @@
+		if err != nil {
+			log.Debugf("Failed to make the path %q relative to the path %q Error %q", irBuildConfig.RepoInfo.TargetPath, irBuildConfig.RepoInfo.GitRepoDir, err)
+		} else {
+			contextPath = filepath.Dir(relDockerfilePath)
+		}
+	}
+	gitRepoURL := irBuildConfig.RepoInfo.GitRepoURL
+	if gitRepoURL == "" {
+		gitRepoURL = gitRepoURLPlaceholder
+	}
+	branchName := irBuildConfig.RepoInfo.GitRepoBranch
+	if branchName == "" {
+		branchName = defaultGitRepoBranch
+	}
+	src := okdbuildv1.BuildSource{}
+	src.Type = okdbuildv1.BuildSourceGit
+	src.Git = &okdbuildv1.GitBuildSource{URI: gitRepoURL, Ref: branchName}
+	src.ContextDir = contextPath
+	src.SourceSecret = &corev1.LocalObjectReference{Name: irBuildConfig.SourceSecretName}
+	return src
+}
+
+func (*BuildConfig) getBuildStrategy(irBuildConfig irtypes.BuildConfig, ir irtypes.EnhancedIR) okdbuildv1.BuildStrategy {
+	dockerfilePath := dockerfilePathPlaceholder
+	if irBuildConfig.RepoInfo.GitRepoDir != "" {
+		relDockerfilePath, err := filepath.Rel(irBuildConfig.RepoInfo.GitRepoDir, irBuildConfig.RepoInfo.TargetPath)
+		if err != nil {
+			log.Debugf("ERROR: Failed to make the path %q relative to the path %q Error %q", irBuildConfig.RepoInfo.TargetPath, irBuildConfig.RepoInfo.GitRepoDir, err)
+		} else {
+			dockerfilePath = relDockerfilePath
+		}
+	}
+	strategy := okdbuildv1.BuildStrategy{}
+	strategy.Type = okdbuildv1.DockerBuildStrategyType
+	strategy.DockerStrategy = &okdbuildv1.DockerBuildStrategy{DockerfilePath: dockerfilePath}
+	return strategy
+}
+
+func (bc *BuildConfig) getBuildTriggerPolicy(irBuildConfig irtypes.BuildConfig, ir irtypes.EnhancedIR) okdbuildv1.BuildTriggerPolicy {
+	gitRepoURL := irBuildConfig.RepoInfo.GitRepoURL
+	if gitRepoURL == "" {
+		gitRepoURL = gitRepoURLPlaceholder
+	}
+	policy := okdbuildv1.BuildTriggerPolicy{Type: bc.getWebHookType(gitRepoURL)}
+	whtrig := okdbuildv1.WebHookTrigger{SecretReference: &okdbuildv1.SecretLocalReference{Name: irBuildConfig.WebhookSecretName}}
+	switch policy.Type {
+	case okdbuildv1.GitHubWebHookBuildTriggerType:
+		policy.GitHubWebHook = &whtrig
+	case okdbuildv1.GitLabWebHookBuildTriggerType:
+		policy.GitLabWebHook = &whtrig
+	case okdbuildv1.BitbucketWebHookBuildTriggerType:
+		policy.BitbucketWebHook = &whtrig
+	default:
+		log.Warnf("the git repo is not on github, gitlab or bitbucket. Using generic webhook")
+		policy.GenericWebHook = &whtrig
+	}
+	return policy
+}
+
+func (*BuildConfig) getWebHookType(gitRepoURL string) okdbuildv1.BuildTriggerType {
+	switch true {
+	case strings.Contains(gitRepoURL, "github"):
+		return okdbuildv1.GitHubWebHookBuildTriggerType
+	case strings.Contains(gitRepoURL, "gitlab"):
+		return okdbuildv1.GitLabWebHookBuildTriggerType
+	case strings.Contains(gitRepoURL, "bitbucket"):
+		return okdbuildv1.BitbucketWebHookBuildTriggerType
+	default:
+		log.Warnf("the git repo is not on github, gitlab or bitbucket. Defaulting to generic web hook trigger")
```

Comment:
Do we want this to be a debug message? Since in many cases the we might not have found a git repo url

Reply:
```Do we have conditions to ensure that if a Openshift cluster (has buildconfig) supports tekton we create only tekton? I did not see any such condition in buildconfig createnewresource.```
https://github.com/HarikrishnanBalagopal/move2kube/blob/buildconfig/internal/transformer/cicdtransformer.go#L39-L50

Reply:
```
Also, do should we create buildconfig and imagestream even if the cluster supports only deployment (means somebody has manually disabled deployment config). This is a corner case, we can handle it as part of sensible defaults (we can create a new issue for it).
```
To get k8s deployments to use image streams we have to add certain annotations https://developers.redhat.com/blog/2019/09/20/using-red-hat-openshift-image-streams-with-kubernetes-deployments/
```
  annotations:
    deployment.kubernetes.io/revision: "2"
    image.openshift.io/triggers: '[{"from":{"kind":"ImageStreamTag","name":"ubi:8"},"fieldPath":"spec.template.spec.containers[?(@.name==\"myapp\")].image"}]'
```

Reply:
> 
> 
> Do we want this to be a debug message? Since in many cases the we might not have found a git repo url

If there's no repo should we just avoid creating CICD resources?

Reply:
We can create it if there is request from customers. For now, create an issue with this link

Reply:
> > 
> > 
> > Do we want this to be a debug message? Since in many cases the we might not have found a git repo url
> 
> If there's no repo should we just avoid creating CICD resources?

Let it be there, with placeholder for git repo.